### PR TITLE
Communication - Added release phone number LRO

### PR DESCRIPTION
**Pull request by jbeauregardb**

### CODE REVIEW

File: sdk/communication/azure-communication-administration/src/test/resources/session-records/beginReleasePhoneNumbers.json [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        "User-Agent" : "azsdk-java-azure-communication-administration/1.0.0-beta.3 (11.0.8; Windows 10; 10.0)"
+      },
+      "Response" : {
+        "Transfer-Encoding" : "chunked",
+        "X-Processing-Time" : "447ms",
+        "MS-CV" : "LA1nngj23EKnZVRmN3kZHQ.0",
+        "retry-after" : "0",
+        "X-Azure-Ref" : "0GyyXXwAAAAANStwPu4xJRJ55d7IxiXsAREVOMDJFREdFMDMwOQA5ZmM3YjUxOS1hOGNjLTRmODktOTM1ZS1jOTE0OGFlMDllODE=",
+        "StatusCode" : "200",
+        "Body" : "{\"releaseId\":\"1e671fd9-31e3-4c93-a688-e3c13e865e25\",\"createdAt\":\"2020-10-26T20:05:46.9508456+00:00\",\"status\":\"Pending\",\"phoneNumberReleaseStatusDetails\":{\"+12134592987\":{\"status\":\"Pending\"}}}",
```
Comment (minnieliu) [inline sdk/communication/azure-communication-administration/src/test/resources/session-records/beginReleasePhoneNumbers.json:34]: Should we be redacting the releaseId and phone number details?

File: sdk/communication/azure-communication-administration/README.md [hunk 2]
```diff
@@ -303,6 +303,22 @@ SyncPoller<Void, Void> res =
...
+    PhoneNumber phoneNumber = new PhoneNumber("PHONE_NUMBER_TO_RELEASE");
+    List<PhoneNumber> phoneNumbers = new ArrayList<>();
+    phoneNumbers.add(phoneNumber);
+    PhoneNumberClient phoneNumberClient = createPhoneNumberClient();
+
+    SyncPoller<PhoneNumberRelease, PhoneNumberRelease> res = 
+        phoneNumberClient.beginReleasePhoneNumbers(phoneNumbers, duration);
+    res.waitForCompletion();
+    PhoneNumberRelease result = res.getFinalResult();
+    System.out.println("Phone number status: " + result.getStatus());
```
Comment (minnieliu) [inline sdk/communication/azure-communication-administration/README.md:318]: Update this to "Phone number release status" ?

Comment (jbeauregardb) [reply sdk/communication/azure-communication-administration/README.md]: Thank you!

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 3]
```diff
@@ -896,4 +896,67 @@
...
+     * @param phoneNumbers A list of {@link PhoneNumber} with the desired numbers to release
+     * @param pollInterval The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the release entity
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<PhoneNumberRelease, PhoneNumberRelease> 
+        beginReleasePhoneNumbers(List<PhoneNumber> phoneNumbers, Duration pollInterval) {
+        Objects.requireNonNull(phoneNumbers, "'phoneNumbers' cannot be null.");
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:914]: Similar to other, can you use the default interval here as well?

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 4]
```diff
@@ -896,4 +896,67 @@
...
+                    String releaseId = releaseNumberResponse.getReleaseId();
+                    Mono<PhoneNumberRelease> phoneNumberRelease = getReleaseById(releaseId);
+                    return phoneNumberRelease;
+                });
+            return response;
+        };
+    }
+
+    private Function<PollingContext<PhoneNumberRelease>, 
+        Mono<PollResponse<PhoneNumberRelease>>>
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:937]: nit: can you move the argument in the above line?

File: sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberAsyncClientIntegrationTest.java [hunk 5]
```diff
@@ -525,6 +494,22 @@ public void beginPurchaseSearch(HttpClient httpClient) {
...
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    public void beginReleasePhoneNumbers(HttpClient httpClient) {
+        PhoneNumber phoneNumber = new PhoneNumber(PHONENUMBER_TO_RELEASE);
+        List<PhoneNumber> phoneNumbers = new ArrayList<>();
+        phoneNumbers.add(phoneNumber);
+        Duration pollInterval = Duration.ofSeconds(5);
+        PollerFlux<PhoneNumberRelease, PhoneNumberRelease> poller =
+            this.getClient(httpClient).beginReleasePhoneNumbers(phoneNumbers, pollInterval);
+        AsyncPollResponse<PhoneNumberRelease, PhoneNumberRelease> asyncRes =
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberAsyncClientIntegrationTest.java:506]: Didnt we say that we will not block and instead use stepVerifier in tests?

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 6]
```diff
@@ -896,4 +896,67 @@
...
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param phoneNumbers A list of {@link PhoneNumber} with the desired numbers to release
+     * @param pollInterval The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the release entity
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<PhoneNumberRelease, PhoneNumberRelease>
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:911]: I am not sure what the developers will understand out of PhoneNumberRelease, can't we have a better naming atleast for this?

Comment (ankitarorabit) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:911]: Ideally, I would want to see something more meaningful -- do we know how other languages are solving this?

**Review by ankitarorabit — CHANGES_REQUESTED** [review]
Added comments

**Review by ankitarorabit — APPROVED** [review]

**Comment by RezaJooyandeh** [comment]
🥳

**Pull request merged**